underfs/tutorial/src/main/java/alluxio/underfs/dummy/DummyUnderFileSystem.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.dummy;
+
+import alluxio.AlluxioURI;
+import alluxio.underfs.BaseUnderFileSystem;
+import alluxio.underfs.UfsDirectoryStatus;
+import alluxio.underfs.UfsFileStatus;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.local.LocalUnderFileSystem;
+import alluxio.underfs.options.CreateOptions;
+import alluxio.underfs.options.DeleteOptions;
+import alluxio.underfs.options.FileLocationOptions;
+import alluxio.underfs.options.MkdirsOptions;
+import alluxio.underfs.options.OpenOptions;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Dummy {@link UnderFileSystem} implementation for tutorial.
+ */
+@ThreadSafe
+public class DummyUnderFileSystem extends BaseUnderFileSystem {
+  public static final String DUMMY_SCHEME = "dummy:/";
```

Comment:
for `DUMMY_SCHEME`, do we require "dummy:/" or "dummy://"?

Reply:
we expect `bin/alluxio fs mount /mnt/dummy dummy://tmp` to write to `/tmp`. excluded the second `/` to not require `dummy:///tmp`

Reply:
file URIs are [required by the spec](https://en.wikipedia.org/wiki/File_URI_scheme#Meaning_of_slash_character) to use `file:///path/`, so we should do the same here

Reply:
@apc999 @aaudiber done, thanks